pkg/agent/openflow/client.go

```diff
@@ -619,8 +619,14 @@ func (c *client) InstallServiceGroup(groupID binding.GroupIDType, withSessionAff
 	c.replayMutex.RLock()
 	defer c.replayMutex.RUnlock()
 
+	var adds, mods []binding.OFEntry
 	group := c.featureService.serviceEndpointGroup(groupID, withSessionAffinity, endpoints...)
-	if err := group.Add(); err != nil {
+	if _, installed := c.featureService.groupCache.Load(groupID); installed {
+		mods = append(mods, group)
+	} else {
+		adds = append(adds, group)
+	}
+	if err := c.bridge.AddOFEntriesInBundle(adds, mods, nil); err != nil {
```

Comment:
`group.Add()` sets `isInstalled`, I don't find `AddOFEntriesInBundle` sets it, which would lead to inconsistent state.

And why it doesn't use `group.Add()`?

Reply:
We will not reply on `isInstalled` anymore. `AddOFEntriesInBundle` can be used to add/modify/delete groups without `isInstalled`. You could see that `groupCache` can be used to decide the command `add` or `modify` of group:

```go
if _, installed := c.featureService.groupCache.Load(groupID); installed {
		mods = append(mods, group)
	} else {
		adds = append(adds, group)
	}
```

Reply:
It would make the code confusing if the field is no longer used but kept there. And why this change?

Reply:
OK, I think we could remove the methods `Add()`, `Modify()` and `Delete()` in `OFEntry` and their corresponding implementations. However, lib ofnet is not only used by Antrea, I think we should not modify the code.

```go
type OFEntry interface {
	Add() error
	Modify() error
	Delete() error
	Type() EntryType
	KeyString() string
	// Reset ensures that the entry is "correct" and that the Add /
	// Modify / Delete methods can be called on this object. This method
	// should be called if a reconnection event happened.
	Reset()
	// GetBundleMessage returns ofctrl.OpenFlowModMessage which can be used in Bundle messages. operation specifies what
	// operation is expected to be taken on the OFEntry.
	GetBundleMessages(operation OFOperation) ([]ofctrl.OpenFlowModMessage, error)
}
```

Reply:
Could you explain why this change?

Reply:
When installing a group with mounts of buckets (assuming more than maximum buckets per message), more than one group mod messages will be generated to install the group. The first group mod message installs the group by `add` command for OVS group. others inserts buckets to the group by `insert_buckets` command for OVS group.

Reply:
I have rewritten method `Add` of group.